src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -357,3 +358,174 @@ def add_noise(
 
     def __len__(self):
         return self.config.num_train_timesteps
+
+
+class DDIMExtendedScheduler(DDIMScheduler):
```

Comment:
We don't want to abstract from existing schedulers, could you maybe instead create a new scheduler file?

You can try to make use of the `#Copied from ...` mechanism: https://github.com/huggingface/diffusers/blob/2bbf8b67a7b6a944ba30033a3c930956f1dcd702/src/diffusers/schedulers/scheduling_lms_discrete.py#L29
for all methods that are exactly the same as DDIM.